src/Umbraco.Infrastructure/HostedServices/WebhookQueueExecuteHostedService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Infrastructure.HostedServices;
+
+public class WebhookQueueExecuteHostedService : BackgroundService
+{
+    private readonly IWebhookBackgroundTaskQueue _taskQueue;
+    private readonly WebhookSettings _webhookSettings;
+
+    public WebhookQueueExecuteHostedService(ILogger<QueuedHostedService> logger, IWebhookBackgroundTaskQueue taskQueue, IOptions<WebhookSettings> webhookSettings)
+    {
+        _taskQueue = taskQueue;
+        _webhookSettings = webhookSettings.Value;
+    }
+
+    private async Task BackgroundProcessing()
+    {
+
+        while (await _taskQueue.DequeueAsync() is Func<Task<WebhookResponseModel>> workItem)
+        {
+            WebhookResponseModel response = await workItem();
+            if (response.HttpResponseMessage.IsSuccessStatusCode || response.RetryCount >= _webhookSettings.MaximumRetries)
+            {
+                continue;
+            }
+
+            response.RetryCount++;
+            await _taskQueue.QueueBackgroundWorkItemAsync(() => RetryWorkItem(workItem, response.RetryCount));
+            await Task.Delay(CalculateDelay(response.RetryCount));
```

Comment:
Problem with delaying here, is that it will wait for this delay here, and not run the other Webhooks.
I feel like there should be a better way to do this, but I cannot think of it 🙈